docs/troubleshooting/clean-install.md

```diff
@@ -0,0 +1,128 @@
+
+### Install Anaconda
+
+If you don't have Anaconda install yet, follow the steps provided on the [Anaconda installation page](https://docs.anaconda.com/anaconda/install/windows/).
+
+### Create a new environment with Streamlit
+
+Next you'll need to set up your environment.
+
+1. Follow the steps provided by Anaconda to [set up and manage your environment](https://docs.anaconda.com/anaconda/navigator/getting-started/#managing-environments) using the Anaconda Navigator.
+
+2. Select the "▶" icon next to your new environment. Then select "Open terminal":
+
+   !["Open terminal" in Anaconda Navigator](https://i.stack.imgur.com/EiiFc.png)
+
+3. In the terminal that appears, type:
+
+   ```sh
+   pip install streamlit
+   ```
+
+4. Test that the installation worked:
+
+   ```sh
+   streamlit hello
+   ```
+
+   Streamlit's Hello app should appear in a new tab in your web browser!
+
+### Use your new environment
+
+1. In Anaconda Navigator, open a terminal in your environment (see step 2 above).
+2. In the terminal that appears, use Streamlit as usual:
+
+   ```sh
+   streamlit run myfile.py
+   ```
+
+## Install Streamlit on macOS/Linux
+
+Streamlit's officially-supported environment manager for macOS and Linux is [Pipenv](https://pypi.org/project/pipenv/). See instructions on how to install and use it below.
+
+### Install Pipenv
+
+1. Install `pip`.
+
+   On a macOS:
+
+   ```sh
+   sudo easy_install pip
+   ```
+
+   On Ubuntu with Python 3:
+   ```sh
+   sudo apt-get install python3-pip
+   ```
+
+   (Or `python-pip` if you're still on Python 2)
+
+   For other Linux distributions, see [How to install PIP for Python](https://www.makeuseof.com/tag/install-pip-for-python/).
+
+2. Install `pipenv`.
+
+   ```sh
+   pip3 install pipenv
+   ```
+
+### Create a new environment with Streamlit
+
+1. Navigate to your project folder:
+
+   ```sh
+   cd myproject
+   ```
+
+2. Create a new Pipenv environment in that folder:
+
+   ```sh
+   pipenv
+   ```
+
+   When you run the command above, a file called `Pipfile` will appear in `myprojects/`. This file is where your Pipenv environment and its dependencies are declared.
+
+3. Install Streamlit in your environment:
+
+   ```sh
+   pip install streamlit
+   ```
+
+   Or if you want to create an easily-reproducible environment, replace `pip` with `pipenv` every time you install something:
+   ```sh
+   pipenv install streamlit
+   ```
+
+4. Test that the installation worked:
+
+   ```sh
+   streamlit hello
+   ```
+
+   Streamlit's Hello app should appear in a new tab in your web browser!
+
+### Use your new environment
+
+1. Any time you want to use the new environment, you first need to go to your project folder (where the `Pipenv` file lives) and run:
+
+   ```sh
+   pipenv shell
+   ```
+
+2. Now you can use Python and Streamlit as usual:
+
+   ```sh
+   streamlit run myfile.py
+   ```
```

Comment:
Is it possible to add a subsection about installation directly from the source code, to get the latest merged features before it gets released? It will be very useful for early adopters.

Reply:
@tvst - Thoughts on this?

Reply:
Let's move this to another PR - I know Thiago is thinking about this separately.

Reply:
Yeah, let's do it as a separate PR. And we can have that section of the docs just point to our [Contributing wiki](https://github.com/streamlit/streamlit/wiki/Contributing).